packages/page-statistics/src/provider/hooks/useProposal.ts

```diff
@@ -1,58 +1,460 @@
-import { ProposalType } from '@webb-dapp/page-statistics/generated/graphql';
-import { Loadable, Page, ProposalStatus } from '@webb-dapp/page-statistics/provider/hooks/types';
+import {
+  ProposalType,
+  useProposalDetailsLazyQuery,
+  useProposalsLazyQuery,
+  useProposalsOverviewLazyQuery,
+  useProposalVotesLazyQuery,
+} from '@webb-dapp/page-statistics/generated/graphql';
+import { mapProposalListItem } from '@webb-dapp/page-statistics/provider/hooks/mappers';
+import { Loadable, Page, PageInfoQuery, ProposalStatus } from '@webb-dapp/page-statistics/provider/hooks/types';
+import { useCurrentMetaData } from '@webb-dapp/page-statistics/provider/hooks/useCurrentMetaData';
+import { useEffect, useMemo, useState } from 'react';
 
+import { Threshold as QueryThreshold } from './types';
+
+/**
+ * Threshold values
+ * @param proposal - The proposer threshold value a session
+ * @param proposers -The number of proposers of a session
+ *
+ * */
 type Thresholds = {
-  proposal: string;
-  proposers: string;
+  proposal: number;
+  proposers: number;
 };
-
+/**
+ * Proposal type Status counter map for proposal statuses
+ * @params accepted - number of accepted proposals
+ * @params rejected - number of rejected proposals
+ * @params open - number of open proposals
+ * @params signed - number of signed proposals
+ * */
 type ProposalTypeStats = {
   accepted: number;
   rejected: number;
   open: number;
   signed: number;
 };
-type ProposalListItem = {
+/** List over view
+ *
+ * A type for infinite number of list will show a defined number of items and will have the count for the full list
+ * @param T - The underlying list item type
+ * @param count - Total number of the list of `T`
+ * */
+type ListOverView<T = string> = {
```

Comment:
```suggestion
type DiscreteList<T = string> = {
```

I would prefer to name this away from "overview" if possible.

Reply:
yeah sounds good, changed it